bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/rest/WebSocketManager.java

```diff
@@ -82,25 +91,49 @@ public String getEvent() {
         }
     }
 
-    public void openSession(@Nullable String sessionToken) throws FreeboxException {
-        WebSocketClient client = new WebSocketClient(apiHandler.getHttpClient());
-        URI uri = getUriBuilder().scheme(getUriBuilder().build().getScheme().contains("s") ? "wss" : "ws").build();
-        ClientUpgradeRequest request = new ClientUpgradeRequest();
-        request.setHeader(ApiHandler.AUTH_HEADER, sessionToken);
-
-        try {
-            client.start();
-            client.connect(this, uri, request);
-        } catch (Exception e) {
-            throw new FreeboxException(e, "Exception connecting websocket client");
+    public void openSession(@Nullable String sessionToken, int reconnectInterval) {
+        if (reconnectInterval > 0) {
+            WebSocketClient client = new WebSocketClient(apiHandler.getHttpClient());
+            URI uri = getUriBuilder().scheme(getUriBuilder().build().getScheme().contains("s") ? "wss" : "ws").build();
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            request.setHeader(ApiHandler.AUTH_HEADER, sessionToken);
+
+            stopReconnect();
+            ScheduledExecutorService scheduler = ThreadPoolManager
+                    .getScheduledPool(FreeboxOsBindingConstants.BINDING_ID);
```

Comment:
How many thread pools will you create by doing that ?
Is it just one for all the binding ?

Reply:
Changed to final member of the class.